crates/net/network/src/cache.rs

```diff
@@ -36,14 +44,33 @@ impl<T: Hash + Eq> LruCache<T> {
         false
     }
 
+    /// Same as [`Self::insert`] but returns a tuple, where the second index is the evicted value,
+    /// if one was evicted.
+    pub fn _insert_with_eviction_feedback(&mut self, entry: T) -> (bool, Option<T>) {
```

Comment:
can we reuse this function in:
`insert{self.insert_evicttion().map(|(inserted,_)| inserted)}`